platform-sdk/swirlds-common/src/main/java/com/swirlds/common/metrics/extensions/PhaseTimerBuilder.java

```diff
@@ -0,0 +1,230 @@
+    private TimeUnit absoluteTimeUnit = UNIT_MICROSECONDS;
+
+    /**
+     * Create a new {@link PhaseTimerBuilder} instance.
+     *
+     * @param platformContext the platform context
+     * @param time            the time provider
+     * @param metricsCategory the metrics category
+     * @param clazz           the enum class that describes the phases
+     */
+    public PhaseTimerBuilder(
+            @NonNull final PlatformContext platformContext,
+            @NonNull final Time time,
+            @NonNull final String metricsCategory,
+            @NonNull final Class<T> clazz) {
+
+        this.clazz = Objects.requireNonNull(clazz);
+        this.platformContext = Objects.requireNonNull(platformContext);
+        this.time = Objects.requireNonNull(time);
+        this.metricsCategory = Objects.requireNonNull(metricsCategory);
+        this.phases = EnumSet.allOf(Objects.requireNonNull(clazz));
+        if (phases.isEmpty()) {
+            throw new IllegalArgumentException("The enum class " + clazz.getName() + " has zero values.");
+        }
+        this.initialPhase = phases.iterator().next();
+    }
+
+    /**
+     * Build the {@link PhaseTimer} instance.
+     *
+     * @return the {@link PhaseTimer} instance
+     */
+    @NonNull
+    public PhaseTimer<T> build() {
+        return new PhaseTimer<>(this);
+    }
+
+    /**
+     * Set the prefix for the metrics names created by this object. If not set, a default is generated.
+     *
+     * @param metricsNamePrefix the prefix for the metrics names created by this object
+     * @return this
+     */
+    @NonNull
+    public PhaseTimerBuilder<T> setMetricsNamePrefix(@NonNull final String metricsNamePrefix) {
+        this.metricsNamePrefix = Objects.requireNonNull(metricsNamePrefix);
+        return this;
+    }
+
+    /**
+     * Set the initial phase. If not set, the phase with ordinal 0 is used.
+     *
+     * @param initialPhase the initial phase
+     * @return this
+     */
+    @NonNull
+    public PhaseTimerBuilder<T> setInitialPhase(@NonNull final T initialPhase) {
+        this.initialPhase = initialPhase;
+        return this;
+    }
+
+    /**
+     * Enable fractional metrics. Disabled by default.
+     *
+     * @return this
+     */
+    @NonNull
+    public PhaseTimerBuilder<T> enableFractionalMetrics() {
+        this.fractionMetricsEnabled = true;
+        return this;
+    }
+
+    /**
+     * Enable absolute time metrics. Disabled by default.
+     *
+     * @return this
+     */
+    @NonNull
+    public PhaseTimerBuilder<T> enableAbsoluteTimeMetrics() {
+        this.absoluteTimeMetricsEnabled = true;
+        return this;
+    }
+
+    /**
+     * Set the unit for absolute time metrics. If not set, microseconds are used.
+     *
+     * @param absoluteTimeUnit the unit for absolute time metrics
+     * @return this
+     */
+    @NonNull
+    public PhaseTimerBuilder<T> setAbsoluteTimeUnit(@NonNull final TimeUnit absoluteTimeUnit) {
```

Comment:
missed opportunity to call this `absoluteUnit`

Reply:
"fixed"